src/bootstrap/builder.rs

```diff
@@ -1237,6 +1237,7 @@ impl<'a> Builder<'a> {
         // efficient initial-exec TLS model. This doesn't work with `dlopen`,
         // so we can't use it by default in general, but we can use it for tools
         // and our own internal libraries.
+        #[cfg(not(target_arch = "powerpc"))]
```

Comment:
I think [`cfg!` macro](https://doc.rust-lang.org/rust-by-example/attribute/cfg.html) would be nicer here.

Reply:
Hm, I'm surprised to see cfg here at all; I'd expect this to check target.contains("powerpc") or so...

Reply:
I believe `cfg` would be wrong when cross-compiling - since "target" here refers to the target of `bootstrap` which would be the host (native) platform?

Reply:
@glaubitz try `if target.triple.startswith("powerpc-") { }` instead

Reply:
Done. I'm testing the change now on a real PowerPC machine to see if it actually works.

Reply:
The change as suggested by @infinity0 works fine and fixes the problem on 32-bit PowerPC.

But please feel free to use a different approach if anyone has a better idea. I don't insist on my particular approach.